core/trino-main/src/main/java/io/trino/connector/informationschema/InformationSchemaPageSource.java

```diff
@@ -272,11 +273,17 @@ private void addTablesRecords(QualifiedTablePrefix prefix)
     {
         Set<SchemaTableName> tables = listTables(session, metadata, accessControl, prefix);
         Set<SchemaTableName> views = listViews(session, metadata, accessControl, prefix);
-        // TODO (https://github.com/trinodb/trino/issues/8207) define a type for materialized views
+        Set<SchemaTableName> materializedViews = listMaterializedViews(session, metadata, accessControl, prefix);
 
-        for (SchemaTableName name : union(tables, views)) {
+        for (SchemaTableName name : union(union(tables, views), materializedViews)) {
             // if table and view names overlap, the view wins
-            String type = views.contains(name) ? "VIEW" : "BASE TABLE";
+            String type = "BASE TABLE";
+            if (materializedViews.contains(name)) {
+                type = "MATERIALIZED VIEW";
```

Comment:
@martint Could you confirm whether returning `MATERIALIZED VIEW` as a table type is okay?

Reply:
No, as it is incompatible with the SQL specification.